### PR TITLE
[feat] Migrate python code to typescript with test code

### PR DESCRIPTION
**Pull request by Pewww**
### Operation
I migrated python code(**py-universal-name-format**) to typescript with test code.
The reason why the coverage of the test code is not 100% is that there is no test-case for the overall function within the 'Name' model.
But, I think it does not matter :D

### Coverage

---

### CODE REVIEW

**Comment by Pewww** [comment]
@k44ng The ones that I'm going to work on after this are listed below. (w.o/ PR)

1. Edit README
2. Publish package

**Review by k4ng — COMMENTED** [review]
Thanks for your pull request. I left some comments for the improvements!

File: src/models/Name.ts [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    if (!Name.isValidName(name)) {
+      throw new Error('Error occured');
+    }
+
+    if (placeholder) {
+      this.name[placeholder] = name;
+    } else {
+      this.name = name;
+    }
+  }
```
Comment (k44ng) [inline src/models/Name.ts:125]: For method chaining, please return the `this`.

File: src/models/Name.ts [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    const prefix = this.getName('prefix');
+    const suffix = this.getName('suffix');
+
+    return `${prefix ? prefix + ' ' : ''}${this.formattedName()}${
+      suffix ? ' ' + suffix : ''
+    }`;
+  }
+
+  public getName(placeholder?: string) {
+    if (placeholder) {
```
[the remark this answers is not in the document]
Comment (Pewww) [reply src/models/Name.ts:104]: Here is the python code.
```
if placeholder:
            if placeholder in self.__name:
                return self.__name[placeholder]
            else:
                return

 else:
     return self.__name
```

If you look at the code, 'this.name' appears to have two types: string and object.
In Python code, type checking is performed once again when the 'placeholder' parameter exists.
However in Typescript, it returns undefined when accessing internal field of 'String' type, so I omitted the checking syntax cause it doesn't have any difference.

[the remark this answers is not in the document]
Comment (Pewww) [reply src/models/Name.ts:104]: Nevertheless, if you think explicit type checking is necessary, please leave a comment! :D

[the remark this answers is not in the document]
Comment (k44ng) [reply src/models/Name.ts:104]: Oh, I get it. It seems good to be resolved. 👍

File: src/models/Name.ts [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+  private order: OrderType = [];
+  private encode: EncodeType = 'western';
+
+  constructor(source: Source) {
+    this.name = source.name;
+    this.order = source.order;
+    this.encode = source.encode;
+  }
+
+  public static isValidName(name: NameType) {
```
[the remark this answers is not in the document]
Comment (Pewww) [reply src/models/Name.ts]: Fixed in 2957a870a0d08feec6ada096af2e60112d4ea144 :D

File: src/models/Name.ts (hunk 1 above)
Comment (Pewww) [reply src/models/Name.ts:127]: Fixed in 2957a870a0d08feec6ada096af2e60112d4ea144 :D

File: src/models/Name.ts [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+      return this;
+    }
+
+    const clone = cloneDeep(this);
+    clone.setName(clone.getName('family').toUpperCase(), 'family');
+
+    return clone;
+  }
+
+  public initialGivenName(isIncludingMiddle?: boolean) {
```
[the remark this answers is not in the document]
Comment (Pewww) [reply src/models/Name.ts]: Fixed it as if we had spoken verbally.

**Comment by k4ng** [comment]
Nice work! It looks good to be merged! 👍

**Review by k4ng — APPROVED** [review]

**Pull request merged**